tests/storage/snapshot.go

```diff
@@ -257,37 +256,27 @@ var _ = SIGDescribe("[Serial]VirtualMachineSnapshot Tests", func() {
 				return vm, vmi
 			}
 
-			checkVMFreeze := func(snapshot *snapshotv1.VirtualMachineSnapshot, vmi *v1.VirtualMachineInstance, shouldFreeze bool) {
+			checkVMFreeze := func(snapshot *snapshotv1.VirtualMachineSnapshot, vmi *v1.VirtualMachineInstance, hasGuestAgent, shouldFreeze bool) {
 				var expectedIndications []snapshotv1.Indication
-				if shouldFreeze {
+				if hasGuestAgent {
 					expectedIndications = []snapshotv1.Indication{snapshotv1.VMSnapshotOnlineSnapshotIndication, snapshotv1.VMSnapshotGuestAgentIndication}
 				} else {
 					expectedIndications = []snapshotv1.Indication{snapshotv1.VMSnapshotOnlineSnapshotIndication, snapshotv1.VMSnapshotNoGuestAgentIndication}
 				}
 				Expect(snapshot.Status.Indications).To(Equal(expectedIndications))
 
-				var conditionsLength int
-				if shouldFreeze {
-					conditionsLength = 3
-				} else {
-					conditionsLength = 2
-				}
+				conditionsLength := 2
 				Expect(snapshot.Status.Conditions).To(HaveLen(conditionsLength))
 				Expect(snapshot.Status.Conditions[0].Type).To(Equal(snapshotv1.ConditionProgressing))
 				Expect(snapshot.Status.Conditions[0].Status).To(Equal(corev1.ConditionFalse))
 				Expect(snapshot.Status.Conditions[1].Type).To(Equal(snapshotv1.ConditionReady))
 				Expect(snapshot.Status.Conditions[1].Status).To(Equal(corev1.ConditionTrue))
+
 				if shouldFreeze {
```

Comment:
this check is now true either way the fsfreezestatus after the snapshot should always be ""

Reply:
good call